### PR TITLE
Feature for matching assessment

### PR DESCRIPTION
**Pull request by ftbernales**
Added the utils for matching assessment. Originally came from smtk but I did not add the entire library. 

_Main_ script to follow

BTW, what's our plan for file management? My idea is just to put everything under `data/`, but probably would be more organized if there would be a `data/input` and `data/output`?

### CODE REVIEW

**Comment by stansays** [comment]
> probably would be more organized if there would be a `data/input` and `data/output`

I agree with this

**Comment by ftbernales** [comment]
But how will that impact the dist files? I don't know how to do that yet so no idea on the pre-reqs. I'm only basing on running our scripts on IDE.

**Comment by stansays** [comment]
Executable should work the same. All dependencies of all the files are imported, and `./` in the scripts will still correspond to wherever the exe file is placed.

**Comment by ftbernales** [comment]
@stansays please help with compiling the code to bundled dist exe, as I haven't done that yet. Are you using _pyinstaller_ for that?

If your time allows, please also help me in fixing bugs here and improving string searches--as commented in some lines. But this will do for our usual naming conventions and practice. I'll just add a README for user guidance.

**Comment by ftbernales** [comment]
BTW, can we consider adding config files for user input/interface? This can apply both to Deepsoil-Compile & matching_assessment.

File: src/matching_assessment.py [hunk 1]
```diff
@@ -0,0 +1,480 @@
...
+    # output_dir = os.path.join(wd, "data/output_files") -> for scripting use
+    output_dir = os.getcwd() + "/output_files" # -> default location for dist
+    build_save_dir(output_dir)
+
+    # Get target response spectra in suite directory
+    ASC_target = import_ASC_target_spectra(input_dir)
+    SZ_target = import_SZ_target_spectra(input_dir)
+
+    # Get RotD1000-Component Response Spectra for ASC Suite
+    suite_rotd100_spectra = compute_suite_rotd100_spectra(input_dir, periods,
```
Comment (ftbernales) [inline src/matching_assessment.py:502]: @stansays Will the mp pool be placed here or in the function definition?

**Comment by stansays** [comment]
> @stansays please help with compiling the code to bundled dist exe, as I haven't done that yet. Are you using _pyinstaller_ for that?
> 
> If your time allows, please also help me in fixing bugs here and improving string searches--as commented in some lines. But this will do for our usual naming conventions and practice. I'll just add a README for user guidance.

I use the command
`pyinstaller -F [script]`

Can the `PeerMSR` class and `hazardlib.geo` module from openquake be removed in `sm_utils.py`, or do we intend to use them? I'm hesitant to compile as of now because of the possibility of ballooning the distribution with dependencies from openquake.

**Comment by ftbernales** [comment]
Done merging the files to [matching_assessment](https://github.com/ftbernales/Deepsoil-Compile/tree/matching_assessment)

**Comment by ftbernales** [comment]
@stansays The bundled code is now working, but can't get it to work in my AMH PC due to admin restriction + reputation-based protection. 

Kindly review one more time. I think we can merge this if you're good with this (_for now_).

**Pull request closed without merging**